File: code/java/剑指offer53-2.java
class Solution {
    public int missingNumber(int[] nums) {
        int l=0,r=nums.length-1;
        while(l<r){
            int mid = (l+r)/2;
            if(nums[mid]==mid) l = mid+1;
            else r = mid-1;
        }
        return nums[l]==l?l+1:l;
    }
}

File: code/python/剑指offer53-2.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        l,r = 0,len(nums)-1
        while l<r:
            mid = (l+r)//2
            if nums[mid]==mid:
                l = mid+1
            else:
                r = mid-1
        if nums[l]==l:
            return l+1
        else:
            return l